lib/kontena/templates/haproxy/http_backends.text.erb

```diff
@@ -0,0 +1,44 @@
+<% services.each do |service| %>
+<% if service.upstreams? %>
+backend <%= service.name %>
+    option forwardfor
+    <% service.custom_settings.each do |setting| %>
+    <%= setting %>
+    <% end %>
+
+    <% if service.cookie %>
+    <% if service.cookie.empty? %>
+    cookie KONTENA_SERVERID insert indirect nocache
+    <% else %>
+    <%= service.cookie %>
+    <% end %>
+    <% end %>
+
+    balance <%= service.balance %>
+
+    <% if service.basic_auth? %>
+    acl auth_ok_<%= service.name %> http_auth(auth_users_<%= service.name %>)
+    http-request auth realm <%= service.name %> unless auth_ok_<%= service.name %>
+    <% end %>
+
+    <% if service.virtual_path? %>
+    <% if !service.keep_virtual_path? %>
+    reqrep ^([^\ :]*)\ <%= service.virtual_path %>[/]?(.*)     \1\ /\2
+    <% end %>
```

Comment:
Somehow makes me feel there should be some `else` branch also? If not, then just combine those if's?

Reply:
This is direct port from current (confd) template. I can combine those if's.